yarpl/include/yarpl/Refcounted.h

```diff
@@ -318,14 +318,32 @@ Reference<CastTo> make_ref(Args&&... args) {
   );
 }
 
-template <typename T>
-Reference<T> get_ref(T* object) {
-  static_assert(
-      std::is_base_of<Refcounted, std::decay_t<T>>::value,
-      "Reference can only be constructed with a Refcounted object");
+class enable_get_ref {
+ private:
+  // force the class to be polymorphic so we can dynamic_cast<T>(this)
+  virtual void dummy_internal_get_ref() {}
```

Comment:
is it true that you need this virtual method only in DEBUG build?
do you trust compiler to optimize out the vtable if the method is not used in RELEASE? or should we add #ifdef?

Reply:
This is a good point, I'll play around with making this conditional on a DEBUG build.